src/Centreon/Domain/Engine/EngineConfiguration.php

```diff
@@ -131,4 +131,28 @@ public static function removeIllegalCharacters(string $stringToAnalyse, ?string
         $illegalCharacters = html_entity_decode($illegalCharacters);
         return str_replace(str_split($illegalCharacters), '', $stringToAnalyse);
     }
+
+    /**
+     * Find all illegal characters from the given string.
+     *
+     * @param string $stringToAnalyse String for which we want to remove illegal characters
+     * @param string|null $illegalCharacters String containing illegal characters
+     * @return bool Return true if illegal characters have been found
+     */
+    public static function hasIllegalCharacters(string $stringToAnalyse, ?string $illegalCharacters): bool
+    {
+        if (null === $illegalCharacters) {
+            return false;
+        }
+        $illegalCharactersList = str_split(html_entity_decode($illegalCharacters), 1);
+        // Add space to the list as, space is not compliant with the RFC
+        $illegalCharactersList[] = ' ';
+
+        foreach ($illegalCharactersList as $illegalCharacter) {
```

Comment:
There is much simpler
`public static function hasIllegalCharacters(string $stringToAnalyse, ?string $illegalCharacters): bool {
    return strlen(stringToAnalyse) !== strlen(EngineConfiguration::removeIllegalCharacters($stringToAnalyse, $illegalCharacters);
}`